esp-hal/src/spi/master.rs

```diff
@@ -1904,18 +1938,19 @@ mod dma {
 
     /// Async functionality
     mod asynch {
-        use core::{
-            cmp::min,
-            ops::{Deref, DerefMut},
-        };
+        #[cfg(any(doc, feature = "unstable"))]
+        use core::cmp::min;
+        use core::ops::{Deref, DerefMut};
 
         use super::*;
 
+        #[cfg_attr(not(feature = "unstable"), allow(dead_code))]
```

Comment:
If this is only used when `unstable` is enabled, shouldn't it just be cfg'd away entirely when the feature is disabled?

Reply:
The problem is that `#[instability::unstable]` marks things `pub(crate)`, and doesn't remove them. If related things are removed completely, the project may end up not compiling.